File: data structures/queue/queue.py
class Queue:
    def __init__(self):
        self.storage = []

    def __len__(self):
        return len(self.storage)

    def enqueue(self, value):
        self.storage.insert(0, value)

    def dequeue(self):
        if len(self.storage) > 0:
            return self.storage.pop()
        else:
            print("Cannot dequeue from empty queue!")
            return None

q = Queue()

q.enqueue(1)
q.enqueue(2)
q.enqueue(3)

print("length of queue:", len(q))
print("items in queue:", q.storage)

print("dequeued value:", q.dequeue())
print("dequeued value:", q.dequeue())
print("dequeued value:", q.dequeue())
print("dequeued value:", q.dequeue())

print("length of queue:", len(q))
print("items in queue:", q.storage)

File: data structures/queue/queue.js
class Queue {
  constructor() {
    this.storage = [];
  }

  length = () => {
    return this.storage.length;
  };

  enqueue = (value) => {
    this.storage.splice(0, 0, value);
  };

  dequeue = () => {
    if (this.storage.length > 0) {
      return this.storage.pop();
    } else {
      console.error('Cannot dequeue from empty queue!');
      return null;
    }
  };
}

const q = new Queue();

q.enqueue(1);
q.enqueue(2);
q.enqueue(3);

console.log('length of queue: ', q.length());
console.log('items in queue: ', q.storage);

console.log('dequeued value: ', q.dequeue());
console.log('dequeued value: ', q.dequeue());
console.log('dequeued value: ', q.dequeue());
console.log('dequeued value: ', q.dequeue());

console.log('length of queue: ', q.length());
console.log('items in queue: ', q.storage);
